dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/config/OpenAPIConfigTest.java

```diff
@@ -255,6 +258,53 @@ public void testApplicationPathConfict() throws Exception {
+        server.updateServerConfiguration(config);
+
+        server.startServer(false);
+
+        assertWebAppStarts("/foo");
+        assertWebAppStarts("/bar");
+
+        assertDocumentPath("/foo");
+        assertUiPath("/bar");
+
+        // Test changed path with various referer headers
+
+        //Default HTTP port with matching docPath
+        OpenAPI model = new OpenAPIConnection(server, "/foo").header("Referer","http://testurl1/foo").downloadModel();
+        //as ports is not a liberty report, only return a single server
+        assertTrue("Check that the list of servers is not empty",model.getServers().size()==1);
+        //check that the server name has changed to allow for either http:// or https:// to be first
+        assertTrue("check that the servers entry use the server from the referer header",model.getServers().get(0).getUrl().contains("http://testurl1/"+APP_NAME));
+
+        //Default HTTPS port with matching uiPath
+        model = new OpenAPIConnection(server, "/foo").header("Referer","https://testurl2/bar").downloadModel();
+        //as ports is not a liberty report, only return a single server
+        assertTrue("Check that the list of servers is not empty",model.getServers().size()==1);
+        //check that the host name has changed to allow for either http:// or https:// to be first
+        assertTrue("check that the servers entry use the server from the referer header",model.getServers().get(0).getUrl().contains("https://testurl2/"+APP_NAME));
+
+        //Rerun Test that if the referer path does not match the endpoints that the original hostname is used when config is not default
+        model = new OpenAPIConnection(server, "/foo").header("Referer","http://testurl3:"+server.getHttpDefaultPort()+"/random/").downloadModel();
+        // Path mismatch, should revert to server host and server http port - https not enabled so only single server should be returned
+        assertTrue("Check that the list of servers has single entries",model.getServers().size() == 1);
+        assertTrue("Check host reverts to the requestUrl host", model.getServers().get(0).getUrl().contains("http://" + server.getHostname()+ ":" + server.getHttpDefaultPort() + "/" + APP_NAME));
+
+        //path does not match, but ends in `/ui`, so should revert to default host, however as port differs from server ports, only a single entry should be being returned
+        model = new OpenAPIConnection(server, "/foo").header("Referer","http://testurl4/random/ui").downloadModel();
+        // Path mismatch with `/ui`, should revert to server host and server http port - https not enabled so only single server should be returned
+        assertTrue("Check that the list of servers has a single entries",model.getServers().size() == 1);
+        assertTrue("Check host reverts to the requestUrl host", model.getServers().get(0).getUrl().contains("http://"+server.getHostname() + ":" + server.getHttpDefaultPort() + "/" + APP_NAME));
```

Comment:
The two comments here are saying that the expected result occurs for different reasons - which is correct?

Reply:
Addressed comments left over from previously larger test case to clarify what we are doing in the tests and the reason for some of our expectations for number of returned servers